lib/slack_rewriter/application/slack_rewriter.ex

```diff
@@ -0,0 +1,88 @@
+defmodule SlackRewriter.Application.SlackRewriter do
+  @moduledoc """
+  SlackRewriter
+  """
+  alias SlackRewriter.Application.UserRegistry
+  alias SlackRewriter.Bridge.Slack
+  alias SlackRewriter.Utils.YoutrackMatcher
+
+  @type response :: {:ok, term()} | {:error, String.t()}
+
+  # add subtype capture using constant
+  # @flat_message_subtypes ["me_message", "file_share"]
+  @nested_message_subtypes ["message_changed", "message_replied"]
+
+  def react_to_message(%{
```

Comment:
queste due firme della `react_to_message` immagino che dipendano dal tipo di messaggio (changed vs replied) però non c'è un modo chiaro per confermarlo. 

Una opportunità potrebbero darcela i tipi (prendo un po' di shortcut)
```elixir
# 
@type message :: message_changed | message_replied | map()

@type message_changed :: %{
        "channel" => String.t(),
        ...
      }

@spec react_to_message(message()) :: response()
```

Reply:
qui in realtà le due firme servono a catturare due strutture diverse che può avere `event`.
La struttura dipende dal tipo di evento e dal suo sottotipo. In questo caso distinguo le due chiamate perché parte delle informazioni, per alcuni message (tra cui il plain o `file_share` etc) la struttura è flat, per altri parte dei campi sono dentro il field `message`.

la clause sulla seconda react_to_message serve a ignorare direttamente categorie di messaggi che non ci interessano.

se ho capito bene il tuo suggerimento è di esplicare meglio questa distinzione anche a livello di typespec, in questo caso forse distinguerei però più tra `flat_message` e `nested_message` dato che poi i `subtype` cui si fa riferimento sono molteplici.

Reply:
si corretto, io ho sparato a caso i nomi dei tipi (proprio perché non si capiva la differenza, e la mia ipotesi era sbagliata 😄

se fossero già esistiti i tipi  `flat_message` e `nested_message` avrei avuto la risposta che cercavo sul "perché servono queste due differenti firme?" 😉